components/DocSearch.tsx

```diff
@@ -27,41 +27,63 @@ const DocSearch: FunctionComponent = () => {
 
-  useEffect(() => {
+  const [isLoading, setLoading] = useState(true);
+
+  const initializeDocSearch = () => {
     let searchHits: SearchHit[] = [];
+    window.docsearch({
+      apiKey: 'db5ac03036725a2cd252824f49ab2b09',
+      indexName: 'avo',
+      inputSelector: '#algolia-avo-search', // the selector of my search input
+      transformData: (hits: SearchHit[]) => {
+        setSearchHits(hits);
+        searchHits = hits;
+      },
+      handleSelected: (
+        input: SearchInput,
+        _: never,
+        suggestion: SearchSuggestion,
+      ) => {
+        setSearchStatus(Status.ResultSelected);
+        const searchString = input.getVal();
+        const searchResultZeroBasedPosition = searchHits
+          .map((hit: SearchHit) => {
+            return hit.url;
+          })
+          .indexOf(suggestion.url);
+        Avo.docsSearchResultSelected({
+          searchResultPosition: searchResultZeroBasedPosition + 1,
+          searchResultCount: searchHits.length,
+          searchTerm: searchString,
+          searchTermCharacterCount: searchString.length,
+        });
+        window.open(suggestion.url, '_self');
+      },
+    });
+    setLoading(false);
+  };
+
+  useEffect(() => {
     if (window.docsearch) {
-      window.docsearch({
-        apiKey: 'db5ac03036725a2cd252824f49ab2b09',
-        indexName: 'avo',
-        inputSelector: '#algolia-avo-search', // the selector of my search input
-        transformData: (hits: SearchHit[]) => {
-          setSearchHits(hits);
-          searchHits = hits;
-        },
-        handleSelected: (
-          input: SearchInput,
-          _: never,
-          suggestion: SearchSuggestion,
-        ) => {
-          setSearchStatus(Status.ResultSelected);
-          const searchString = input.getVal();
-          const searchResultZeroBasedPosition = searchHits
-            .map((hit: SearchHit) => {
-              return hit.url;
-            })
-            .indexOf(suggestion.url);
-          Avo.docsSearchResultSelected({
-            searchResultPosition: searchResultZeroBasedPosition + 1,
-            searchResultCount: searchHits.length,
-            searchTerm: searchString,
-            searchTermCharacterCount: searchString.length,
-          });
-          window.open(suggestion.url, '_self');
-        },
-      });
+      initializeDocSearch();
     } else {
       // eslint-disable-next-line no-console
-      console.warn('Docsearch has failed to load');
+      console.info(
+        'Still waiting for Docsearch to load, setting up onload listener...',
+      );
+      const scriptTag = document.querySelector('#load-docsearch-script');
+      if (scriptTag) {
+        scriptTag.addEventListener('load', function () {
+          // eslint-disable-next-line no-console
+          console.info('Docsearch loaded, initializing...');
+          initializeDocSearch();
+        });
+      } else {
+        // eslint-disable-next-line no-console
```

Comment:
I'd like to keep these logs in if we need to debug this in the future. In 90% of the cases we'll never get in here